tests/crypto/base_n_tests.cpp

```diff
@@ -76,17 +76,14 @@ static void test_58( const std::string& test, const std::string& expected )
 
     std::vector<char> dec = fc::from_base58( enc1 );
     BOOST_CHECK_EQUAL( vec.size(), dec.size() );
-    BOOST_CHECK( !memcmp( vec.data(), dec.data(), vec.size() ) );
+    BOOST_CHECK( vec == dec );
 
     char buffer[64];
     size_t len = fc::from_base58( enc1, buffer, 64 );
     BOOST_CHECK( len <= 64 );
-    BOOST_CHECK( !memcmp( vec.data(), buffer, len ) );
+    BOOST_CHECK( vec.empty() || !memcmp( vec.data(), buffer, len ) );
     if ( len > 10 ) {
-        try {
-            len = fc::from_base58( enc1, buffer, 10 );
-            BOOST_CHECK( len <= 10 );
-        } catch ( fc::exception expected ) {}
+        BOOST_CHECK_THROW(fc::from_base58( enc1, buffer, 10 ), fc::exception);
```

Comment:
Why this change? It's totally different logic.

Reply:
Actually the test expects fc::from_base58 to throw. We're just using the standard Boost way rather than a quite obscure try{ BOOST_CHECK(); } catch(){}

In essence  it's exactly the same test.

Reply:
The old test code will still pass if `fc::from_base58` doesn't throw. In this case will check `len <= 10`. So this is different test. I don't know whether the old test code is buggy though.